runtime/schema.js

```diff
@@ -116,10 +116,23 @@ export default class Schema {
       return schema2;
     else if (schema2.isMoreSpecificThan(schema1))
       return schema1;
-    
-    // TODO: Don't be lazy
-    assert(false, 'non-trivial intersection of schemas not implemented.');
-    return null;
+
+    let names = [...schema1.names].filter(name => schema2.names.includes(name));
+    let fields = {};
+
+    for (let [field, type] of Object.entries(schema1.fields)) {
+      let otherType = schema2.fields[field];
+      if (otherType && Schema.typesEqual(type, otherType)) {
+        fields[field] = type;
+      }
+    }
+
+    if (Object.keys(fields).length === 0) return null;
```

Comment:
I think it's OK to return an object with no name and no fields, instead of null here.

Reply:
Fixed.